src/common/exif.cc

```diff
@@ -3225,7 +3512,9 @@ int dt_exif_xmp_attach(const int imgid, const char *filename)
 
     // last but not least attach what we have in DB to the XMP. in theory that should be
     // the same as what we just copied over from the sidecar file, but you never know ...
-    dt_exif_xmp_read_data(xmpData, imgid);
```

Comment:
Not visible in the diff here, can you tell me what is happening for dt_exif_xmp_read_data(). The header is removed, is the body still needed?

Reply:
> The header is removed, is the body still needed?
Sorry I don't understand this ...

> what is happening for dt_exif_xmp_read_data()

That's the routine which gathers all metadata from dt database, plus history of masks and plugins into xmpData.
It was used both to write xmp companion file and to include metadata in exported files.
But the metadata can, but not have to be the same in both cases.
XMP companion has to mirror the database (not to loose anything)
Export file has to transmit the information needed by destination.

dt_exif_xmp_read_data_export() is a copy of dt_exif_xmp_read_data() except it calls the tag routines for exportation. And logically other metadata should be filterable in the same manner.
But history is useless for exportation, so I suggest to remove it from dt_exif_xmp_read_data_export().
So both routines are needed.

Reply:
Ok, I see, thanks for the explanations.

Reply:
> But history is useless for exportation, so I suggest to remove it from dt_exif_xmp_read_data_export().

It's not useless, so such change has to be discussed separately.

Reply:
Maybe a confusion, are you saying that you are removing the xmp which is embedded in JPEG (at exportation time)?

Reply:
> It's not useless,

HI @paraffin, thanks for jumping in. What the utility of history in exported images?

Reply:
It's a documented feature, see "load sidecar file" button description in the manual: https://darktable.gitlab.io/doc/en/history_stack.html

Reply:
OK, clear. Thanks. So we have to let history there. To keep or not the history can be an option amongst the others to control the metadata to be exported.